tsec-libsodium/src/test/scala/tsec/libsodium/KeyExchangeTest.scala

```diff
@@ -0,0 +1,87 @@
+package tsec.libsodium
+
+import cats.effect.IO
+import tsec.libsodium.cipher.{CryptoSecretBox, PlainText}
+import tsec.libsodium.kx._
+import tsec.common._
+
+class KeyExchangeTest extends SodiumSpec {
+
+  behavior of "Sodium KeyExchange"
+
+  it should "encrypt and decrypt properly" in {
+    forAll { (s: String) =>
+      val plainText = PlainText(s.utf8Bytes)
+
+      val program: IO[PlainText] = for {
+        server <- KeyExchange.generateKeyPair[IO]
+        client <- KeyExchange.generateKeyPair[IO]
+
+        clientSession <- KeyExchange.generateClientSessionKeys[IO](client, server.pubKey)
+        serverSession <- KeyExchange.generateServerSessionKeys[IO](server, client.pubKey)
+
+        // client to server
+        clientKey1 <- CryptoSecretBox.buildKey[IO](clientSession.send)
+        serverKey1 <- CryptoSecretBox.buildKey[IO](serverSession.receive)
+        enc1       <- CryptoSecretBox.encrypt[IO](plainText, clientKey1)
+        dec1       <- CryptoSecretBox.decrypt[IO](enc1, serverKey1)
+
+        // server to client
+        clientKey2 <- CryptoSecretBox.buildKey[IO](clientSession.receive)
+        serverKey2 <- CryptoSecretBox.buildKey[IO](serverSession.send)
+        enc2       <- CryptoSecretBox.encrypt[IO](dec1, serverKey2)
+        dec2       <- CryptoSecretBox.decrypt[IO](enc2, clientKey2)
+
+      } yield dec2
+
+      program.unsafeRunSync() mustBe plainText
```

Comment:
Array[Byte] equality is hairy and fickle since it may possibly be by reference (I'm very unsure whether we have strong guarantees on same-element equality in scalacheck.. I just don't trust em).

`program.unsafeRunSync().toUtf8String mustBe s` or comparing them both as hex strings seems like the proper thing to do.